src/app/utils/file-upload.helper.ts

```diff
@@ -0,0 +1,41 @@
+import { ToastrService } from 'ngx-toastr';
+
+export function validateAndAppendFiles(
+  formData: FormData,
+  files: File[],
+  fileType: string,
+  allowedExtensions: string[],
+  maxSizeInMB: number,
+  toastr: ToastrService,
+): { allFilesValid: boolean; needsUpdate: boolean } {
+  let needsUpdate = false;
+  let allFilesValid = true;
```

Comment:
instead of declaring two variables here. Should have an error object which would contain or the different errors based on the type. 

If the error object is empty (doesn't have any key Object.keys(errors) === 0) we would consider that all the fields are valid